app/controllers/reports_controller.rb

```diff
@@ -122,6 +136,14 @@ def add_data_to_sheet_for_export_applications(sheet, applications)
     end
   end
 
+  def getCountsForUserSignInReport(one_week_where_clause, two_weeks_where_clause, under_a_month_where_clause, over_a_month_where_clause)
+    @counts = {}
+    @counts["One Week"] = User.where(one_week_where_clause).count
+    @counts["Two Weeks"] = User.where(two_weeks_where_clause).count
+    @counts["Under a month"] = User.where(under_a_month_where_clause).count
+    @counts["Over a month"] = User.where(over_a_month_where_clause).count
+  end
```

Comment:
You could save some typing here by using a hash literal:

    @counts = { :"One Week" => User.where(one_week_where_clause).count,
                        :"Two Weeks" => User.where....,
                        :"Under a month" => ..... ,
                        :" Over a month" => ..... , }

Or if you want to use symbols for keys instead of strings:

    @counts = { :one_week => User.where(one_week_where_clause).count,
                        :two_weeks => User.where....,
                        :under_a_month => ..... ,
                        :over_a_month => ..... , }

or using the new syntax for symbols:

    @counts = { one_week: User.where(one_week_where_clause).count,
                        two_weeks: User.where....,
                        under_a_month: ..... ,
                        over_a_month: ..... , }